File: ext/lynx/cache/driver.zep.c

#ifdef HAVE_CONFIG_H
#include "../../ext_config.h"
#endif

#include <php.h>
#include "../../php_ext.h"
#include "../../ext.h"

#include <Zend/zend_exceptions.h>

#include "kernel/main.h"


ZEPHIR_INIT_CLASS(Lynx_Cache_Driver) {

	ZEPHIR_REGISTER_INTERFACE(Lynx\\Cache, Driver, lynx, cache_driver, lynx_cache_driver_method_entry);

	return SUCCESS;

}

/**
 * To initialize cache provider
 */
ZEPHIR_DOC_METHOD(Lynx_Cache_Driver, getInstance);

/**
 * Save data
 */
ZEPHIR_DOC_METHOD(Lynx_Cache_Driver, save);

/**
 * Save data by id
 */
ZEPHIR_DOC_METHOD(Lynx_Cache_Driver, delete);

/**
 * Fetch data by id
 */
ZEPHIR_DOC_METHOD(Lynx_Cache_Driver, get);

/**
 * Is row exist?
 */
ZEPHIR_DOC_METHOD(Lynx_Cache_Driver, exists);

File: ide/0.0.3/Lynx/cache/Driver.zep.php
<?php

namespace Lynx\Cache;

interface Driver
{

    /**
     * To initialize cache provider
     * 	
     */
	public function getInstance();

    /**
     * Save data
     * 	
     *
     * @param string $id 
     * @param mixed $data 
     * @param mixed $lifeTime 
     */
	public function save($id, $data, $lifeTime = 3600);

    /**
     * Save data by id
     * 	
     *
     * @param string $id 
     */
	public function delete($id);

    /**
     * Fetch data by id
     * 	
     *
     * @param string $id 
     */
	public function get($id);

    /**
     * Is value exist?
     * 	
     *
     * @param string $id 
     */
	public function exists($id);

    /**
     * Flush all values
     * 	
     */
	public function flush();

}
